rtengine/simpleprocess.cc

```diff
@@ -1737,21 +1730,24 @@ class ImageProcessor
             readyImg = tempImage;
         }
 
+        MetadataInfo info(imgsrc->getFileName());
         switch (params.metadata.mode) {
-            case MetaDataParams::TUNNEL:
-                // Sending back the whole first root, which won't necessarily be the selected frame number
-                // and may contain subframe depending on initial raw's hierarchy
-                readyImg->setMetadata(initialImage->getMetaData()->getRootExifData());
-                break;
-
-            case MetaDataParams::EDIT:
-                // ask for the correct frame number, but may contain subframe depending on initial raw's hierarchy
-                readyImg->setMetadata(initialImage->getMetaData()->getBestExifData(imgsrc, &params.raw), params.exif, params.iptc);
-                break;
-
-            default: // case MetaDataParams::STRIP
-                // nothing to do
-                break;
+        case MetaDataParams::TUNNEL:
+            // Sending back the whole first root, which won't necessarily be the selected frame number
+            // and may contain subframe depending on initial raw's hierarchy
+            // readyImg->setMetadata (ii->getMetaData()->getRootExifData ());
+            readyImg->setMetadata(std::move(info));
+            break;
+        case MetaDataParams::EDIT:
+            info.setExif(params.exif);
+            info.setIptc(params.iptc);
+            readyImg->setMetadata(std::move(info));
+            // ask for the correct frame number, but may contain subframe depending on initial raw's hierarchy
+            // readyImg->setMetadata (ii->getMetaData()->getBestExifData(imgsrc, &params.raw), params.exif, params.iptc);
```

Comment:
Remove the commented code or update the variable name.
```suggestion
            // Sending back the whole first root, which won't necessarily be the selected frame number
            // and may contain subframe depending on initial raw's hierarchy
            // readyImg->setMetadata (initialImage->getMetaData()->getRootExifData ());
            readyImg->setMetadata(std::move(info));
            break;
        case MetaDataParams::EDIT:
            info.setExif(params.exif);
            info.setIptc(params.iptc);
            readyImg->setMetadata(std::move(info));
            // ask for the correct frame number, but may contain subframe depending on initial raw's hierarchy
            // readyImg->setMetadata (initialImage->getMetaData()->getBestExifData(imgsrc, &params.raw), params.exif, params.iptc);
```

Reply:
Done in #6631